### PR TITLE
First Ansible Roles API test

### PR DESCRIPTION
**Pull request by sambible**
Requires https://github.com/SatelliteQE/nailgun/pull/841

First test surrounding Ansible Roles

### CODE REVIEW

**Review by adarshdubey-star — CHANGES_REQUESTED** [review]
Please add :expectedresults: as well

File: tests/foreman/api/test_ansible.py [hunk 1]
```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
...
     assert len(task_details[0].output['result']['created']) == playbooks_count
+
+
+def test_positive_assign_ansible_role(target_sat):
+    """
+    Test API for assigning an Ansible Role to a host
+
+    :id: 95d0213e-a5a9-4671-8a9e-89a77fc67f39
+
+    :customerscenario: false
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_ansible.py:70]: :customerscenario: is only required when it is true, please remove

File: tests/foreman/api/test_ansible.py [hunk 2]
```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
...
+
+    :steps:
+
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_ansible.py:73]:
```suggestion
```

File: tests/foreman/api/test_ansible.py [hunk 3]
```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
...
+    :CaseAutomation: Automated
+    """
+    SELECTED_ROLE = 'RedHatInsights.insights-client'
+    host = target_sat.api.Host().create()
+    id = target_sat.api.SmartProxy(name=target_sat.hostname).search()[0].id
+    target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
+    host.assign_roles(data={'ansible_role_ids': [1]})
+    host_roles = host.list_roles()
+    assert host_roles[0]['name'] == SELECTED_ROLE
```
Comment (Griffin-Sullivan) [inline tests/foreman/api/test_ansible.py:87]: If we have a list of all roles, we could turn most of this into a fixture and parametrize the test with the list of roles.

**Comment by Griffin-Sullivan** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

File: tests/foreman/api/test_ansible.py [hunk 4]
```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
...
+        3. Assign that role to a host
+        4. Assert that the role was assigned to the host successfully
+
+    :CaseAutomation: Automated
+    """
+    SELECTED_ROLE = 'RedHatInsights.insights-client'
+    host = target_sat.api.Host().create()
+    id = target_sat.api.SmartProxy(name=target_sat.hostname).search()[0].id
+    target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
+    host.assign_roles(data={'ansible_role_ids': [1]})
```
Comment (Griffin-Sullivan) [inline tests/foreman/api/test_ansible.py:85]:
```suggestion
    host.assign_role(data={'ansible_role_ids': [1]})
```

File: tests/foreman/api/test_ansible.py (hunk 3 above)
Comment (sambible) [reply tests/foreman/api/test_ansible.py:87]: Hmm, I do like that idea
This is just the first of the available roles on any given satellite. Maybe make a constant list of those, and then do as you suggest

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

File: tests/foreman/api/test_ansible.py (hunk 4 above)
Comment (sambible) [reply tests/foreman/api/test_ansible.py:82]: ah, yeah had to update my nailgun pr - changed the name to be consistent with the behavior

**Review by Griffin-Sullivan — APPROVED** [review]

File: tests/foreman/api/test_ansible.py [hunk 5]
```diff
@@ -59,3 +59,26 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
...
+    """
+    Test API for assigning an Ansible Role to a host
+
+    :id: 95d0213e-a5a9-4671-8a9e-89a77fc67f39
+
+    :steps:
+        1. Create a Host
+        2. Import a roles into satellite
+        3. Assign that role to a host
+        4. Assert that the role was assigned to the host successfully
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_ansible.py:74]: Ack, only pending @adarshdubey-star 's comment

File: tests/foreman/api/test_ansible.py [hunk 6]
```diff
@@ -59,3 +59,26 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
...
+
+    :CaseAutomation: Automated
+    """
+    SELECTED_ROLE = 'RedHatInsights.insights-client'
+    host = target_sat.api.Host().create()
+    id = target_sat.api.SmartProxy(name=target_sat.hostname).search()[0].id
```
Comment (ogajduse) [inline tests/foreman/api/test_ansible.py:80]:
```suggestion
    id = target_sat.internal_capsule.id
```

**Review by ogajduse — CHANGES_REQUESTED** [review]

**Review by ColeHiggins2 — APPROVED** [review]

**Review by pondrejk — APPROVED** [review]

**Comment by adarshdubey-star** [comment]
@sambible any updates on this PR?

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

File: tests/foreman/api/test_ansible.py (hunk 5 above)
Comment (Gauravtalreja1) [reply tests/foreman/api/test_ansible.py:74]: @sambible this is pending since 2 months, could you pls address it?

**Comment by sambible** [comment]
Addressed comments and implemented @ogajduse change request. Going to run the test again with runner, but should be good to go when we can merge the Nailgun PR

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

File: tests/foreman/api/test_ansible.py [hunk 7]
```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
...
+    :expectedresults:
+        1. Host should be assigned the proper role.
+
+    :CaseAutomation: Automated
+    """
+    SELECTED_ROLE = 'RedHatInsights.insights-client'
+    host = target_sat.api.Host().create()
+    id = target_sat.internal_capsule.id
+    target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
+    host.assign_roles(data={'ansible_role_ids': [1]})
```
Comment (adarshdubey-star) [inline tests/foreman/api/test_ansible.py:85]:
```suggestion
    host.assign_ansible_roles(data={'ansible_role_ids': [1]})
```

File: tests/foreman/api/test_ansible.py [hunk 8]
```diff
@@ -59,3 +59,29 @@ def test_fetch_and_sync_ansible_playbooks(target_sat):
...
+    host_roles = host.list_roles()
```
Comment (adarshdubey-star) [inline tests/foreman/api/test_ansible.py:86]:
```suggestion
    host_roles = host.list_ansible_roles()
```

File: tests/foreman/api/test_ansible.py (hunk 7 above)
Comment (adarshdubey-star) [reply tests/foreman/api/test_ansible.py:85]: I think this is the cause of the PRT failure seeing you nailgun PR

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

**Review by adarshdubey-star — APPROVED** [review]

**Comment by devendra104** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'
nailgun: 841

**Comment by sambible** [comment]
Nailgun PR is merged, this can go In now.

**Comment by Griffin-Sullivan** [comment]
@sambible rebase and we can merge!

**Comment by Griffin-Sullivan** [comment]
I guess we need to rebase again

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k 'test_positive_assign_ansible_role'

File: tests/foreman/api/test_ansible.py (hunk 6 above)
Comment (ogajduse) [reply tests/foreman/api/test_ansible.py]: @sambible Oops, apologies. It should be either `nailgun_capsule`, or `nailgun_smart_proxy`.

**Comment by JacobCallahan** [comment]
@sambible Can you check on the PRT failure and revisit this PR? It would be nice to have this merged.

**Comment by Griffin-Sullivan** [comment]
I vote we just close this PR as [tests/foreman/api/test_ansible.py::test_positive_ansible_job_on_host](https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/api/test_ansible.py#L70) has already covered what this test is doing. There are also equivalent tests in CLI and UI now, so I don't see the need to keep this test when it is already set up for those other tests.

**Pull request closed without merging**